pip/_vendor/pkg_resources/__init__.py

```diff
@@ -2941,10 +2941,10 @@ def _get_mro(cls):
     """Get an mro for a type or classic class"""
     if not isinstance(cls, type):
 
-        class cls(cls, object):
+        class new_cls(cls, object):
             pass
 
-        return cls.__mro__[1:]
+        return new_cls.__mro__[1:]
```

Comment:
> You would expect that change to get overwritten the next time the vendored library is updated.

Umm... It's been added to a patch that pip can apply on the vendored package. So, it would get updated but it's still fragile because the patch if the line numbers change.